infra/charts/turing/subcharts/merlin/templates/deployment.yaml

```diff
@@ -0,0 +1,216 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "merlin.fullname" .}}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ include "merlin.name" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    helm.sh/chart: {{ include "merlin.chart" . }}
+    app.kubernetes.io/name: {{ include "merlin.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}    
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      app: {{ template "merlin.name" .}}
+      release: {{ .Release.Name }}
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 4
+      maxUnavailable: 0
+  template:
+    metadata:
+      labels:
+        app: {{ template "merlin.name" .}}
+        release: {{ .Release.Name }}
+      annotations:
+        prometheus.io/scrape: 'true'
+        prometheus.io/port: "{{ .Values.service.internalPort }}"
+    spec:
+      {{- if and .Values.vault.enabled .Values.environmentConfigs }}
+      initContainers:
+        {{- range $cluster := .Values.environmentConfigs }}
+        - name: "vault-dummy-{{ $cluster.cluster }}-cluster-creds"
+          image: curlimages/curl:latest
+          command: ['/bin/sh', '-c']
+          args:
+            - |
+              curl --fail -v \
+                -H "X-Vault-Token: root" \
+                -d '{"master_ip": "kubernetes:443", "certs":"", "client_certificate":"","client_key":""}' \
+                "{{ template "merlin.vault.address" $ }}/v1/secret/{{ $cluster.cluster }}"
+        {{- end }}
+      {{- end }}
+      containers:
+      - name: merlin
+        image: "{{ .Values.image.registry }}/{{ .Values.image.repository }}:{{ .Values.image.tag }}"
```

Comment:
I wonder if this pattern should be used, we are guilty of being inconsistent here. See [this](https://github.com/gojek/turing/blob/main/infra/charts/turing/templates/deployment.yaml#L60). Sometimes we need the forward slash and sometimes we dont need it. I wonder if one day we should standardise this and do what other helm repositories do; [just repository and tag](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml#L40). I don't think it should be addressed in this PR but maybe a comment indicating the forward slash requirement/non requirement.

Reply:
This _IS_ an attempt to standardize it. Among three (mlp/merlin/turing), only Turing expected to have a trailing slash as part of the registry name. So the link you included (See [this](https://github.com/gojek/turing/blob/main/infra/charts/turing/templates/deployment.yaml#L60)) was actually updated in this PR to address the inconsistency.

Reply:
Ok, I'm still in the middle of reviewing this PR so I have not reached this part yet. But ok glad this was addressed.